app/gui/src/project-view/stores/visualization/index.ts

```diff
@@ -103,40 +104,47 @@ export const [provideVisualizationStore, useVisualizationStore] = createContextS
       })
     }
 
+    const loadedScripts = new Set<string>()
+    const loadedStyles = new Set<string>()
     const scriptsNode = document.head.appendChild(document.createElement('div'))
     scriptsNode.classList.add('visualization-scripts')
-    const loadedScripts = new Set<string>()
-    function loadScripts(module: VisualizationModule) {
+
+    function loadAsync(
+      urls: string[],
+      type: 'scripts' | 'styles',
+      container: HTMLElement,
+      loaded: Set<string>,
+    ) {
       const promises: Promise<void>[] = []
-      if ('scripts' in module && module.scripts) {
-        if (!Array.isArray(module.scripts)) {
-          console.warn('Visualiation scripts should be an array:', module.scripts)
-        }
-        const scripts = Array.isArray(module.scripts) ? module.scripts : [module.scripts]
-        for (const url of scripts) {
-          if (typeof url !== 'string') {
-            console.warn('Visualization script should be a string, skipping URL:', url)
-          } else if (!loadedScripts.has(url)) {
-            loadedScripts.add(url)
-            const node = document.createElement('script')
+      for (const url of urls) {
+        if (!loaded.has(url)) {
+          loaded.add(url)
+          const nodeKind = type === 'scripts' ? 'script' : 'link'
+          const node = document.createElement(nodeKind)
+          if (type === 'styles') {
+            assert(node instanceof HTMLLinkElement)
+            node.href = url
+            node.rel = 'stylesheet'
+          } else if (type === 'scripts') {
+            assert(node instanceof HTMLScriptElement)
```

Comment:
This should be a console error and a graceful `continue` instead of an assert. The visualizations code needs to be treated as user input.

Reply:
The `node` here is created a few lines above depending on the `type` variable, and the assert plays the role of a safe type coercion.